src/components/ChatContent/ToolsContent.tsx

```diff
@@ -223,3 +227,200 @@ export const ToolContent: React.FC<{
     </Container>
   );
 };
+
+export type ToolContentProps = {
+  toolCalls: ToolCall[];
+};
+
+export const ToolContent: React.FC<ToolContentProps> = ({ toolCalls }) => {
+  const ids = toolCalls.reduce<string[]>((acc, cur) => {
+    if (cur.id !== undefined) return [...acc, cur.id];
+    return acc;
+  }, []);
+  const allToolResults = useAppSelector(selectManyToolResultsByIds(ids));
+
+  return processToolCalls(toolCalls, allToolResults);
+};
+
+function processToolCalls(
+  toolCalls: ToolCall[],
+  toolResults: ToolResult[],
+  processed: React.ReactNode[] = [],
+) {
+  if (toolCalls.length === 0) return processed;
+  const [head, ...tail] = toolCalls;
+  const result = toolResults.find((result) => result.tool_call_id === head.id);
+
+  if (result && isMultiModalToolResult(result)) {
+    const restInTail = takeWhile(tail, (toolCall) => {
+      const nextResult = toolResults.find(
+        (res) => res.tool_call_id === toolCall.id,
+      );
+      return nextResult !== undefined && isMultiModalToolResult(nextResult);
+    });
+
+    const nextTail = tail.slice(restInTail.length);
+    const multiModalToolCalls = [head, ...restInTail];
+    const ids = multiModalToolCalls.map((d) => d.id);
+    const multiModalToolResults: MultiModalToolResult[] = toolResults
+      .filter(isMultiModalToolResult)
+      .filter((toolResult) => ids.includes(toolResult.tool_call_id));
+
+    const elem = (
+      <MultiModalToolContent
```

Comment:
Also, here might be better to include `key` prop also, since we are iterating an array of `toolCalls`

Reply:
Added